### PR TITLE
new: Supports MinIO as alternative to AWS S3

### PR DESCRIPTION
**Pull request by fandigunawan**
Needs an alternative S3 compatible file storage as it is against their policy. The widely used S3 compatible system includes [MinIO](https://min.io/). MinIO uses PHP AWS S3 SDK as pointed in MISP Composer requirements and MinIO's documentation.

#### What does it do?

Supports MinIO as an alternative to AWS S3 for file storage #3560
Check file whether it is exists, if it is not exists it will throw exception that will notify upper layer code.

#### Questions

- [ ] Does it require a DB change?
- [X] Are you using it in production?
- [ ] Does it require a change in the API (PyMISP for example)?

### CODE REVIEW

File: app/Lib/Tools/AWSS3Client.php [hunk 1]
```diff
@@ -39,10 +42,39 @@ private function __getSetSettings()
...
+                 //   - https://docs.guzzlephp.org/en/5.3/clients.html#verify
+                 // Example: 
+                 // -- Verify certificate
+                 //    'http'    => ['verify' => '/usr/lib/ssl/certs/minio.pem'],
+                 // -- Do not verify certificate, securitywise, this option is not recommended, however due to 
+                 //    internal deployment scheme it is acceptable risk to set this to false
+                 //    'http'    => ['verify' => false],
+                 // -- Verify againts  built in CA certificates
+                 //    'http'    => ['verify' => true],
+                 'http'    => ['verify' => false],
```
Comment (JakubOnderka) [inline app/Lib/Tools/AWSS3Client.php]: Thats dangerous and maybe makes sense in your env, but it should not be default.

Comment (fandigunawan) [reply app/Lib/Tools/AWSS3Client.php]: I will make it default to trust internal trust store

Comment (adulau) [reply app/Lib/Tools/AWSS3Client.php]: @fandigunawan Can you update the pull-request? Like that we can merge it. Thanks a lot.

Comment (fandigunawan) [reply app/Lib/Tools/AWSS3Client.php]: Sorry, I was very busy lately.
I have pushed the patch to default the TLS validation to true and adds custom trusted CA file.

Comment (adulau) [reply app/Lib/Tools/AWSS3Client.php]: Thank you

**Comment by adulau** [comment]
Thank you for the contribution and the updates!

**Pull request merged**